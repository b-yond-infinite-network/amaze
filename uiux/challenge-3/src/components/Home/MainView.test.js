import MainView from './MainView';
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router'
import configureMockStore from 'redux-mock-store'
import middlewares from './../../store'

const initialState = {
    home: {
        tags: ['tag1', 'tag2']
    },
    common: {
        appName: 'test appName',
        token: '1234'
    }
};

describe('MainView Component', () => {

    let mockStore;

    let tree;
    let wrapper;
    let render;

    let mockOnTabClick;

    beforeAll(() => {

        mockStore = configureMockStore(middlewares)(initialState);
        mockOnTabClick = jest.fn();

        const props = {
            mockOnTabClick: mockOnTabClick
        };

        tree = <Provider store={mockStore}>
            <MemoryRouter initialEntries={["/"]}>
                <MainView {...props} />
            </MemoryRouter>
        </Provider>;
        wrapper = mount(tree);
        console.log(wrapper.debug());
        render = renderer.create(tree).toJSON();

    });

    it('Should match the snapshot', () => {
        expect(render).toMatchSnapshot();
    });

    it('Should render without errors', () => {
        const component = wrapper.find('#main-view');
        expect(component.length).toBe(1);
    });

});