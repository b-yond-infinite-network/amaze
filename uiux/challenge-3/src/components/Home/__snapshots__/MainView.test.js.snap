// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainView Component Should match the snapshot 1`] = `
<div
  className="col-md-9"
  id="main-view"
>
  <div
    className="feed-toggle"
  >
    <ul
      className="nav nav-pills outline-active"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href=""
          onClick={[Function]}
        >
          Your Feed
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href=""
          onClick={[Function]}
        >
          Global Feed
        </a>
      </li>
    </ul>
  </div>
  <div
    className="article-preview"
  >
    Loading...
  </div>
</div>
`;
